src/utils/zcl_abapgit_time.clas.testclasses.abap

```diff
@@ -0,0 +1,100 @@
+CLASS ltcl_time_test DEFINITION FINAL
+  FOR TESTING
+  DURATION SHORT
+  RISK LEVEL HARMLESS.
+
+  PRIVATE SECTION.
+    METHODS get_unix FOR TESTING.
+    METHODS get_utc FOR TESTING.
+ENDCLASS.
+
+
+CLASS ltcl_time_test IMPLEMENTATION.
+
+  METHOD get_unix.
+
+    DATA: lv_unix    TYPE zcl_abapgit_time=>ty_unixtime,
+          lv_tz      TYPE tznzone,
+          lv_utcdiff TYPE tznutcdiff,
+          lv_utcsign TYPE tznutcsign.
+
+    CALL FUNCTION 'TZON_GET_OS_TIMEZONE'
+      IMPORTING
+        ef_timezone = lv_tz.
+
+    CALL FUNCTION 'TZON_GET_OFFSET'
+      EXPORTING
+        if_timezone      = lv_tz
+        if_local_date    = sy-datum
+        if_local_time    = sy-uzeit
+      IMPORTING
+        ef_utcdiff       = lv_utcdiff
+        ef_utcsign       = lv_utcsign
+      EXCEPTIONS
+        conversion_error = 1
+        OTHERS           = 2.
+
+    TRY.
+        lv_unix = '1574605521'.
+        lv_unix+11 = lv_utcsign.
+        lv_unix+12 = lv_utcdiff.
+
+        cl_abap_unit_assert=>assert_equals(
+            act                  = zcl_abapgit_time=>get_unix( iv_date = '20191124'
+                                                               iv_time = '152521' )
+            exp                  = lv_unix ). " User-specific test!
+
+      CATCH cx_root.
```

Comment:
suggest changing the method definition to be `RAISING cx_root`, then these 4 statements can be removed, and the code is fully focused on the functionallity

Reply:
Can't define cx_root as raising exception: The class "CX_ROOT" was not derived from either "CX_STATIC_CHECK" or "CX_DYNAMIC_CHECK". Chose cx_static_check instead.